### PR TITLE
feat(opencode-wellknown): browser plugin + lean frontend primary; per-agent tool-injection model (ADR-0048)

### PR DESCRIPTION
**Pull request by stephane-segning**
## 1. Summary

This PR reworks the org-wide opencode `.well-known` (`charts/librechat-opencode-wellknown`) to give every user a real-browser capability and a leaner, safer agent topology:

- **Add `@vymalo/opencode-browser`** (maintainer-authored, `0.7.0` line) org-wide — `page` + `control` groups only (`debug`/`browser_eval` dropped). Bumps the `@vymalo/*` plugin line `0.6.3 → 0.7.0`.
- **Remove the local `chrome-devtools` MCP** — the plugin's `browser_*` tools replace it; all remaining MCP servers are now remote (`/mcp/*`).
- **Promote `frontend` to the org-wide default PRIMARY** (`config.default_agent`) but keep it **lean** (edit + JS toolchain only); it delegates tool-heavy work to subagents (`@browser`, `@design`, `@doc-research`).
- **Vision-only model pinning** — pin a multimodal model **only** on agents whose tools return images (`@browser` → `browser_screenshot`, `@design` → `refero_get_screen_image`); every other agent (primary + all roles) carries no `model` and inherits the user's session model.
- **Guardrail** — don't also connect the dual package's MCP form (would double-register the 27 `browser_*` tools).
- **[ADR-0048](docs/adr/0048-global-browser-plugin-and-per-agent-tool-injection.md)** documents all of the above, and **corrects ADR-0044's token rationale**.

It solves:

- Source of truth: **[ADR-0048](docs/adr/0048-global-browser-plugin-and-per-agent-tool-injection.md)** (the documented decision) and the upstream plugin **https://github.com/vymalo/opencode-oauth2/tree/main/packages/opencode-browser**. No tracked issue — this is maintainer-directed work, recorded as an ADR per repo convention.

---

## 2. Intent

The intent of this PR is:

> Give opencode users a real browser to close the UI loop (implement → reload → screenshot → decide → iterate), without bloating the default agent's context. While wiring it, we verified opencode's actual tool-injection behaviour and found a load-bearing correction to ADR-0044: opencode **does** scope the *injected* tool set per agent (`session/llm/request.ts` `resolveTools` → `Permission.disabled`: a string-form `*`-deny drops a tool from `tools`/`activeTools`; built-in/MCP/plugin alike; an agent's `allow` wins via `findLast`). That makes the deny-baseline a real per-agent token lever — so the lean primary never carries the 27 `browser_*` schemas, only `@browser` does (on delegation).

---

### In Scope

- `charts/librechat-opencode-wellknown/values.yaml` — plugin entry, `default_agent`, deny-baseline, the `frontend` primary + `@browser`/`@design` subagents, model rule.
- `charts/ai-models/values.yaml` — refreshed `adorsys-frontend` multimodal rationale comment.
- Docs: new `docs/adr/0048-*`, `docs/adr/README.md` + `docs/arc42.md` §9 index rows, `docs/opencode-well-known.md` (agent table + per-agent injection note).

### Out of Scope

- No release tag cut — this is shippable via `tools/release.sh` separately when desired.
- No browser **MCP** form connected (deliberately; the plugin is the single source).
- Per-role cost-lean model pins are **dropped** (the new vision-only rule) — re-pinning a cheap model per role is left as a follow-up if cost beats simplicity.

---

## 4. Verification

I verified this change by:

- [x] Running automated tests (`helm lint --strict` + `helm template --dry-run` per CI logic)
- [x] Running manual tests (rendered the well-known JSON and asserted the agent/model/permission structure)
- [x] Checking logs (traced opencode 1.17.6 source to confirm the per-agent tool-injection behaviour)
- [ ] Checking metrics
- [x] Testing error cases (confirmed render is clean; no browser MCP double-registration)
- [x] Testing permissions/security behavior (deny-baseline keeps `browser_*` out of every agent except `@browser`)
- [ ] Testing rollback or failure behavior, if relevant

Commands run:

```bash
helm dep build charts/librechat-opencode-wellknown/
helm lint charts/librechat-opencode-wellknown/ --strict && helm template x charts/librechat-opencode-wellknown/ --dry-run
helm lint charts/ai-models --strict && helm template x charts/ai-models --dry-run
# render + assert agent/model wiring
helm template wk charts/librechat-opencode-wellknown/ | python3 -c '...'  # see Results
```

Results:

```text
wellknown OK            # strict lint + render
ai-models OK            # orchestrator render
default_agent: frontend | frontend mode: primary | frontend tools: ['bash', 'edit']
agents and model pins:
  browser  -> camer-digital/adorsys-frontend   (multimodal, pinned)
  design   -> camer-digital/adorsys-frontend   (multimodal, pinned)
  frontend -> INHERIT
  web-search/doc-research/iac/reviewer/test/skill -> INHERIT
```

---

## 5. Screenshots / Evidence

- Decision record: [ADR-0048](docs/adr/0048-global-browser-plugin-and-per-agent-tool-injection.md)
- How-to / contract: [docs/opencode-well-known.md](docs/opencode-well-known.md)
- opencode source evidence: `tool/registry.ts` (full catalog), `session/llm/request.ts` `resolveTools` + `permission/index.ts` `disabled`/`fromConfig` (per-agent injection filter), `session/llm.ts` (`activeTools = prepared.tools`).

---

## 6. Risk Assessment

Risk level:

* [ ] Low
* [x] Medium
* [ ] High

Potential risks:

* Org-wide opencode config: `default_agent: frontend` makes every user *start* on the frontend primary (user config still wins on merge).
* Dropping per-role cost-lean pins: high-volume roles (search/review/…) now run on the user's session model — a possible cost regression vs the old `adorsys-*` pins.
* The browser plugin is **local** (bridge + companion extension on the user's machine, Node ≥22); a user without that setup sees the `browser_*` tools fail to connect.

Mitigation:

* All three are documented (ADR-0048 consequences + chart comments). `default_agent` is overridable per user; cheap-model pins are a one-line re-add per role; the local-plugin trade-off matches the removed chrome-devtools MCP.
* Pure config change rendered by Helm; rollback = revert. No release tag is cut by this PR.

---

## 7. AI Usage Declaration

AI was used for:

* [x] Understanding existing code
* [x] Generating code
* [x] Refactoring
* [ ] Generating tests
* [x] Drafting documentation
* [x] Reviewing the diff
* [ ] Not used

Human verification:

* [x] I understand every meaningful change in this PR
* [x] I checked generated code manually
* [ ] I checked generated tests manually
* [x] I removed unsupported AI assumptions
* [x] I accept responsibility for this PR

---

## 8. Reviewer Focus

Please focus your review on:

* [x] Correctness
* [x] Architecture
* [x] Security
* [ ] Performance
* [ ] Tests
* [x] Maintainability
* [x] Product intent
* [ ] Edge cases

> Specifically: (1) the per-agent tool-injection claim in ADR-0048 (worth confirming on a live opencode — delegate to a role and check the primary lacks `browser_*`); (2) whether forcing `default_agent: frontend` org-wide is desired; (3) the cost trade-off of dropping per-role model pins.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: charts/librechat-opencode-wellknown/values.yaml [hunk 1]
```diff
@@ -44,21 +52,72 @@ wellKnown:
...
+      # context every turn.
+      #
+      # ⚠️ Do NOT ALSO add a browser server to `mcp:` below. opencode-browser is
+      # a DUAL package — the same `browser_*` catalog ships as this plugin AND as
+      # an MCP server. Connecting both registers the 27 tools TWICE (same names),
+      # doubling them in every agent that allows `browser_*`. The PLUGIN is the
+      # right form here: it's local-only (bridge + extension on the user's box,
+      # so it can't ride the remote gateway /mcp/* routes anyway) and its opencode
+      # adapter saves screenshots to disk for the model to read. One source only.
+      - - "@vymalo/opencode-browser"
```
[the remark this answers is not in the document]
Comment (stephane-segning) [reply charts/librechat-opencode-wellknown/values.yaml]: Good catch — fixed in the latest push: the tuple now pins `@vymalo/opencode-browser@0.7.0`, so all four `@vymalo/*` plugins are pinned in lockstep and the browser tool surface can no longer drift independently.
